Two tar fixes

ixdy opened the pull request:
I have no idea how this ever worked:
- Bazel's `pkg_tar` defaults `mode = "0555"`, which means that directories inside the tarball are not writable, which causes problems during extraction. If we override that default to `""`, we can let `build_tar` choose an appropriate default. (The upstream `build_tar` also has this functionality, even.)
- When creating parent directories, our `build_tar` used the mode of whatever was being added, which meant the directories might be missing the `x` or `w` bit. I blatantly stole the upstream approach to set the `x` bit when the `r` bit is set, and have also ensured that all parent directories have user mode RWX.

/assign @mikedanese @fejta @BenTheElder

fejta commented on defs/pkg.bzl:
```diff
@@ -22,4 +22,11 @@ load(
 def pkg_tar(**kwargs):
```
`def pkg_tar(mode = "", build_tar = "@io_k8s_repo_infra//tools/build_tar, **kw)`

ixdy commented:
added one more compatibility fix - it seems like the upstream `build_tar` uses 0755 for all extra directories it adds, so I've updated our `build_tar` to ensure that the user mode on directories always has RWX as well.

fejta commented:
/lgtm

mikedanese commented:
Thanks.

/lgtm

ixdy commented:
/hold cancel
